zipline/data/data_portal.py

```diff
@@ -648,38 +648,52 @@ def get_adjusted_value(self, asset, field, dt,
     def _get_minute_spot_value(self, asset, column, dt, ffill=False):
         reader = self._get_pricing_reader('minute')
 
-        if ffill:
+        if not ffill:
+            try:
+                return reader.get_value(asset.sid, dt, column)
+            except NoDataOnDate:
+                if column != 'volume':
+                    return np.nan
+                else:
+                    return 0
+        else:
+            try:
+                # Optimize the best case scenario of a liquid asset
+                # returning a valid price.
+                result = reader.get_value(asset.sid, dt, column)
+                if column != 'volume':
```

Comment:
Since the body of all of these checks is the same, could we coalesce all the conditionals here into something like:

```
    is_valid = (
        (column != 'volume' and not pd.isnull(result)) or
        (result == 0)
    )
    if is_valid:
        return result
```

Better yet, since we don't support `ffill` for non-price fields, could this just be:

```python
assert column == 'price'  # might be 'close'?
if not pd.isnull(result):
    return result
```

Reply:
You are right. We only ever `ffill` the `close`, turning it into `price`. Removing the checks for `volume` once the function reaches the `ffill` is `True` state.